rmw_fastrtps_shared_cpp/test/test_rmw_init_options.cpp

```diff
@@ -0,0 +1,198 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gtest/gtest.h"
+
+#include "rcutils/allocator.h"
+#include "rcutils/strdup.h"
+#include "rmw/init_options.h"
+#include "rmw/security_options.h"
+#include "rmw/types.h"
+
+#include "osrf_testing_tools_cpp/scope_exit.hpp"
+
+#include "rmw_fastrtps_shared_cpp/rmw_init.hpp"
+
+using rmw_fastrtps_shared_cpp::rmw_init_options_init;
+using rmw_fastrtps_shared_cpp::rmw_init_options_copy;
+using rmw_fastrtps_shared_cpp::rmw_init_options_fini;
+
+TEST(RMWInitOptionsTest, init_w_invalid_args_fails) {
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  EXPECT_EQ(
```

Comment:
For expected failure cases here and below, can you add a short comment describing what error you're trying to catch?

Reply:
See https://github.com/ros2/rmw_fastrtps/pull/385/commits/c879be7bde8609876f533870290051d6bb96b1d3